ddl/ddl_api.go

```diff
@@ -658,6 +658,42 @@ func checkColumnsAttributes(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+// checkColumnFieldLength check the maximum length limit for different character set varchar type columns.
+func checkColumnFieldLength(schema *model.DBInfo, colDefs []*ast.ColumnDef, tbInfo *model.TableInfo) error {
+	for _, colDef := range colDefs {
+		if colDef.Tp.Tp == mysql.TypeVarchar {
+			var setCharset string
```

Comment:
should we modify the tbInfo.Charset if it is empty?

Reply:
If `tbInfo.Charset` is empty, tidb will not modify it.